Orange/widgets/visualize/owsieve.py

```diff
@@ -210,23 +150,55 @@ def showEvent(self, ev):
         OWWidget.showEvent(self, ev)
         self.updateGraph()
 
-    ## updateGraph - gets called every time the graph has to be updated
+    def reset_selection(self):
+        self.selection = set()
+        self.update_selection()
+
+    def select_area(self, area, ev):
+        if ev.button() != Qt.LeftButton:
+            return
+        index = self.areas.index(area)
+        if ev.modifiers() & Qt.ControlModifier:
+            self.selection ^= {index}
+        else:
+            self.selection = {index}
+        self.update_selection()
+
+    def update_selection(self):
+        if self.areas is None:
+            self.send("Selection", None)
+            return
+
+        selected = np.zeros(len(self.data), dtype=bool)
```

Comment:
@lanzagar I couldn't use the standard filters (at least I don't know of any that can do this), so this doesn't work on SQL. It should be trivial to add it to the query, though.